pkg/apiserver/resthandler.go

```diff
@@ -488,8 +488,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		gv := scope.Kind.GroupVersion()
 		codec := runtime.NewCodec(
 			scope.Serializer.EncoderForVersion(s, gv),
-			scope.Serializer.DecoderToVersion(s, unversioned.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
-		)
+			scope.Serializer.DecoderToVersion(s, gv))
```

Comment:
This is wrong. We need to encode to the advertised group version for this handler so we can apply the patch, but then decode back to the internal version, since that's what rest handlers deal with internally